parachain/parachain.star

```diff
@@ -75,19 +44,27 @@ def start_local_parachain_node(plan, args, parachain_config, para_id):
                 "-c",
                 "{0} --base-path=/tmp/{1} --chain=/build/{1}-raw.json --ws-port=9946 --port=30333 --rpc-port=9933 --ws-external --rpc-external --prometheus-external --rpc-cors=all --{2} --collator --rpc-methods=unsafe --force-authoring --execution=wasm -- --chain=/app/raw-polkadot.json --execution=wasm".format(binary, chain_name, node["name"]),
             ]
-        parachain_spawn_detail = spawn_parachain(plan, "{0}-{1}-{2}".format(parachain, node["name"].lower(), args["chain-type"]), image, exec_comexec_commandmand, build_file = raw_service.name)
+        
+        build_file = raw_service.name
+        parachain_spawn_detail = node_setup.spawn_parachain(plan, node["prometheus"], "{0}-{1}-{2}".format(parachain, node["name"].lower(), args["chain-type"]), image, exec_comexec_commandmand, build_file, rpc_port_value, prometheus_port_value, lib2lib_port_value)
         parachain_detail["service_name"] = parachain_spawn_detail.name
         parachain_detail["endpoint"] = utils.get_service_url("ws", parachain_spawn_detail.ip_address, parachain_spawn_detail.ports["ws"].number)
-        parachain_detail["endpoint_prometheus"] = utils.get_service_url("tcp", parachain_spawn_detail.ip_address, parachain_spawn_detail.ports["metrics"].number)
-        parachain_detail["service_name"] = parachain_spawn_detail.name
-        parachain_detail["prometheus_port"] = parachain_spawn_detail.ports["metrics"].number
         parachain_detail["ip_address"] = parachain_spawn_detail.ip_address
         parachain_detail["prometheus"] = node["prometheus"]
         parachain_detail["node-type"] = node["node-type"]
+        if node["prometheus"] == True:
+            parachain_detail["prometheus_port"] = parachain_spawn_detail.ports["metrics"].number
+        if prometheus_port_value != None or prometheus_port_value != 0:
+            parachain_detail["prometheus_public_port"] = prometheus_port_value
+            parachain_detail["endpoint_prometheus"] = utils.get_service_url("tcp", "127.0.0.1", prometheus_port_value)
+        if rpc_port_value != None:
+            parachain_detail["endpoint_public"] = utils.get_service_url("ws", "127.0.0.1", rpc_port_value)
+
         parachain_final[parachain_spawn_detail.name] = parachain_detail
+
     return parachain_final
 
-def start_nodes(plan, args, relay_chain_ip):
+def start_nodes(plan, args, relay_chain_ip, rpc_port = [[]], prometheus_port = [[]], lib2lib_port = [[]]):
```

Comment:
@shreyasbhat0 , please review the logic behind the port passing in start_nodes function